quarkus/tests/integration/src/main/java/org/keycloak/it/utils/RawKeycloakDistribution.java

```diff
@@ -0,0 +1,326 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.it.utils;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSession;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+import org.apache.commons.io.FileUtils;
+import org.jboss.logging.Logger;
+
+import io.quarkus.bootstrap.util.ZipUtils;
+import org.keycloak.common.Version;
+
+public final class RawKeycloakDistribution implements KeycloakDistribution {
+
+    private static final Logger LOGGER = Logger.getLogger(RawKeycloakDistribution.class);
+
+    private Process keycloak;
+    private int exitCode = -1;
+    private final Path distPath;
+    private final List<String> outputStream = new ArrayList<>();
+    private final List<String> errorStream = new ArrayList<>();
+    private boolean manualStop;
+    private String relativePath;
+    private int httpPort;
+    private boolean debug;
+    private ExecutorService outputExecutor;
+
+    public RawKeycloakDistribution(boolean debug, boolean manualStop) {
+        distPath = prepareDistribution();
```

Comment:
We should prepareDistribution only after setting the `reCreate` flag that now is removed from this PR. Could we please revert this change?